plugins/inputs/opensearch_query/opensearch_query_test.go

```diff
@@ -0,0 +1,774 @@
+package opensearch_query
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchutil"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	servicePort = "9200"
+	testindex   = "test-opensearch"
+)
+
+type osAggregationQueryTest struct {
+	queryName                 string
+	testAggregationQueryInput osAggregation
+	expectedMetrics           []telegraf.Metric
+	wantBuildQueryErr         bool
+	wantGetMetricFieldsErr    bool
+	wantQueryResErr           bool
+	wantInitErr               bool
+}
+
+var queryPeriod = config.Duration(time.Second * 600)
+
+var testOpensearchAggregationData = []osAggregationQueryTest{
+	{
```

Comment:
Please define those in the test instead of making it a global variable!

Reply:
These tests are used in multiple tests.  Should I wrap the tests in a function?

Reply:
These tests are used in multiple test functions.  Should I refactor this to be a function that returns the test data instead of having it as a global?

Reply:
Yeah that would be better. Or move it to a file and read that. My concern with the code as-is is that you are prone to side-effects if one test (accidentally) modifies the variable...

Reply:
I broke this out to a function that returns the test set.